File: VueApp/gerenciador-projetos/src/domain/projeto/Projeto.js
export default class Projeto {

    constructor(projetoID = 0, descricao = '', resumo = '', dataInicial = '', dataFinal = '', situacaoProjeto = { situacaoProjetoID: 0, descricao: '' }) {

        this.projetoID = projetoID;
        this.descricao = descricao;
        this.resumo = resumo;
        this.dataInicial = dataInicial;
        this.dataFinal = dataFinal;
        this.situacaoProjeto = situacaoProjeto;
    }

    projetoDTO(projeto) {
        this.id = projeto.projetoID;
        this.projetoID = projeto.projetoID;
        this.descricao = projeto.descricao;
        this.resumo = projeto.resumo;
        this.dataInicial = projeto.dataInicial;
        this.dataFinal = projeto.dataFinal;
        this.situacaoProjeto.situacaoProjetoID = projeto.situacaoProjeto.situacaoProjetoID;
    }
}

File: WebAPI/GerenciadorProjetos/Models/Planejamento/Projeto.cs
using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;
using System.Linq;
using System.Threading.Tasks;

namespace GerenciadorProjetos.Models.Planejamento
{
    [Table("Projeto")]
    public class Projeto
    {
        public int ProjetoID  { get; set; }

        [Required]
        [MaxLength(200), MinLength(3)]
        public string Descricao { get; set; }

        public string Resumo { get; set; }

        [Required]
        public DateTime DataInicial { get; set; }

        public DateTime? DataFinal { get; set; }

        public int SituacaoProjetoID { get; set; }
        public SituacaoProjeto SituacaoProjeto { get; set; }

        public ICollection<Atividade> Atividades { get; set; }
    }
}
